content/en/docs/setup/install/multicluster/common.sh

```diff
@@ -77,36 +77,15 @@ function verify_load_balancing()
   (KUBECONFIG="${KUBECONFIG_FILES[1]}"; _wait_for_deployment sample helloworld-v2)
   (KUBECONFIG="${KUBECONFIG_FILES[1]}"; _wait_for_deployment sample sleep)
 
-  # Verify that traffic is load balanced from both clusters.
-  _verify_lb_with_load_function snip_verifying_crosscluster_traffic_1
-  _verify_lb_with_load_function snip_verifying_crosscluster_traffic_3
-}
-
-# _verify_lb_with_load_function that traffic for a given function is balanced
-# between the two clusters
-#
-# $1: the function to be called to generate traffic.
-function _verify_lb_with_load_function()
-{
-  local CALL_HELLOWORLD="$1"
-  local PREV_OUTPUT=""
-  local LOAD_BALANCED=""
-  for i in {1..20}; do
-    OUTPUT="$($CALL_HELLOWORLD)"
-    echo "HelloWorld output: $OUTPUT"
-
-    if [[ -n "$PREV_OUTPUT" ]] && [[ "$PREV_OUTPUT" != "$OUTPUT" ]]; then
-      LOAD_BALANCED="true"
-      break
-    fi
-    PREV_OUTPUT="$OUTPUT"
+  local EXPECTED_RESPONSE_FROM_CLUSTER1="Hello version: v1, instance:"
+  local EXPECTED_RESPONSE_FROM_CLUSTER2="Hello version: v2, instance:"
 
-    # Wait before trying again
-    sleep 2
-  done
+  # Verify we hit both clusters from CLUSTER1
+  VERIFY_RETRIES=9
+  _verify_contains snip_verifying_crosscluster_traffic_1 "$EXPECTED_RESPONSE_FROM_CLUSTER1"
+  _verify_contains snip_verifying_crosscluster_traffic_1 "$EXPECTED_RESPONSE_FROM_CLUSTER2"
```

Comment:
This approach should work, but it could be slow because retries back off exponentially. The first retry delay is 1 sec, the next is 2 sec, 4 sec, 8 sec, 16 sec ... so if it takes all 9 retries before you get what you're looking for, the total delay will be more than 5 minutes I think. That's why I suggested just doing 10 quick curls and make sure you get at least 1 of each version in the output.

Reply:
It seems fine in my runs so far, and it's simpler. I'd say let's keep it as-is until we see its'a an issue.